### PR TITLE
Show fee in a custom asset for OperationDetails module

### PR DESCRIPTION
**Pull request by svojsu**
- Parse the transfer extrinsic to find the 'AssetTxPaymentPallet.assetTxFeePaidEvent' event
- Use the 'assetId' from the event as the fee asset
- Adapt the 'OperationDetails' module to show the fee in a custom asset

### CODE REVIEW

File: novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Fee.swift [hunk 1]
```diff
@@ -35,25 +47,54 @@ extension ExtrinsicProcessor {
         }
     }
 
+    private func findTransactionFeePaidInCustomAsset(
```
Comment (ERussel) [inline novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Fee.swift:50]: I think we can reuse existing function ExtrinsicProcessor.findFeeInCustomAsset

File: novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Matching.swift [hunk 2]
```diff
@@ -665,4 +675,25 @@ extension ExtrinsicProcessor {
             return nil
         }
     }
+
+    private func localAsset(
```
Comment (ERussel) [inline novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Matching.swift:679]: method name look to general. Probably we can move it to AssetHubTokensConverter

File: novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Matching.swift [hunk 3]
```diff
@@ -90,6 +90,11 @@ extension ExtrinsicProcessor {
                 runtimeJsonContext: context
             )
 
+            let feeAsset = localAsset(
```
Comment (ERussel) [inline novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Matching.swift:93]: Here we need to have orml specific parsing. More precisely the parser must match the chain and apply Hydra specific parsing.

**Review by ERussel — COMMENTED** [review]
Overall I wouldn't merge it now since we don't have subquery ready for the custom fees.

File: novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Fee.swift [hunk 4]
```diff
@@ -35,25 +40,85 @@ extension ExtrinsicProcessor {
...
+            guard
+                let feeAssetId = chain.utilityChainAssetId()?.assetId,
+                let nativeFee = optNativeFee?.amount
+            else {
+                return nil
+            }
+
+            return Fee(
+                amount: nativeFee,
+                assetId: feeAssetId
```
Comment (ERussel) [inline novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Fee.swift:94]: In other place, I saw we returning nil for native fee

**Review by ERussel — APPROVED** [review]

File: novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Fee.swift (hunk 4 above)
Comment (svojsu) [reply novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Fee.swift:94]: Oh, I will check before the merge. Thanks.

**Pull request merged**